### ISSUE TITLE
Setting 'term' does nothing

### ISSUE THREAD

**Issue by lifepillar** [macvim-dev/macvim#879]
By default, `:set term?` in MacVim returns `builtin_gui`, and `echo $TERM` in an terminal window returns `xterm`. I can manually  change the value of `$TERM` to `xterm-256color` from within a terminal by typing `TERM=xterm-256color`. But putting `set term=xterm-256color` in my `vimrc` does not have any effect. And trying to set `'term'` at runtime results in `E530: Cannot change term in GUI`.

Is there a way to set `$TERM` automatically?

**Comment by eirnym** [comment]
GUI has it's own terminal emulator, so you can't set it. If you run MacVim in Terminal, `term` option working as you expect.

**Comment by lifepillar** [comment]
Thanks, I was not aware about such difference.

I use my own terminfo file (`~/.terminfo/78/xterm-256color`), which enables italics in the terminal. If I launch MacVim (by double-clicking) and open a terminal window, italics is not enabled in that window because `TERM` is set to `xterm` and cannot be modified by MacVim. I can work around this issue by opening MacVim from Terminal.app as you suggest or by symlinking `~/.terminfo/78/xterm` to `~/.terminfo/78/xterm-256color`. In either case, my terminfo file appears to be loaded, as italics is enabled. If there is no other way, I'll go with the symlink (not sure whether that has side effects, though).

But I am wondering: since MacVim's terminal emulator supports (more than) 256 colors, why not setting `TERM` to `xterm-256color` by default?

**Comment by ychin** [comment]
This is not MacVim specific, as the terminal emulator is a Vim feature. See `:h terminal-unix` for more details:

    Unix ~
    							*terminal-unix*
    On Unix a pty is used to make it possible to run all kinds of commands.  You
    can even run Vim in the terminal!  That's used for debugging, see below.
    
    Environment variables are used to pass information to the running job:
        TERM		the name of the terminal, from the 'term' option or
    			$TERM in the GUI; falls back to "xterm" if it does not
    			start with "xterm"
        ROWS		number of rows in the terminal initially
        LINES		same as ROWS
        COLUMNS		number of columns in the terminal initially
        COLORS		number of colors, 't_Co' (256*256*256 in the GUI)
        VIM_SERVERNAME	v:servername
        VIM_TERMINAL	v:version

Basically, just do `let $TERM='xterm-256color'` before you launch the terminal emulator.

**Comment by lifepillar** [comment]
Thanks, that does it!
